PhysicsTools/PatAlgos/plugins/PATMuonProducer.cc

```diff
@@ -188,8 +201,92 @@ PATMuonProducer::~PATMuonProducer()
 {
 }
 
+GlobalPoint* PATMuonProducer::getMuonDirection(const reco::MuonChamberMatch& chamberMatch,
```

Comment:
this can be a ```unique_ptr``` to avoid the need to keep track of the lifetime

Reply:
fixed